src/main/java/org/assertj/assertions/generator/BaseAssertionGenerator.java

```diff
@@ -362,12 +406,23 @@ private String generateAssertionEntryPointMethodsFor(final Set<ClassDescription>
       String assertionEntryPointMethodContent = assertionEntryPointMethodTemplate.getContent();
       // resolve class assert (ex: PlayerAssert)
       // in case of inner classes like Movie.PublicCategory, class assert will be MoviePublicCategoryAssert
+      String customAssertionClass = fullyQualifiedAssertClassName(classDescription);
+      if (resolvedImports.contains(customAssertionClass)) {
+        // if the FQN exists in the resolved imports then we can use its simple class name instead
+        customAssertionClass = simpleAssertClassName(classDescription);
+      }
       assertionEntryPointMethodContent = replace(assertionEntryPointMethodContent, CUSTOM_ASSERTION_CLASS,
-                                                 fullyQualifiedAssertClassName(classDescription));
+                                                 customAssertionClass);
       // resolve class (ex: Player)
       // in case of inner classes like Movie.PublicCategory use class name with outer class i.e. Movie.PublicCategory.
+      String classToAssert = classDescription.getFullyQualifiedClassName();
+      if (resolvedImports.contains(classDescription.getPackageName() + "." + classDescription.getOuterClassName())) {
```

Comment:
use the `classDescription.getOuterClassFullyQualifiedName()` (see previous comment)